### PR TITLE
Added Rake support

### PR DESCRIPTION
**Pull request by bkuhlmann**
## Overview

The following adds Rake support so it's easier to wire up this gem using Rake and run code quality analysis from the CLI (or wired up via CI).

## Details

- See commits for details.

## Notes

Something is off with the way FakeFS works where the full RSpec test suite won't run with this new spec. You can only verify this works locally by running:

``` 
bundle exec rspec spec/lib/rake/task_spec.rb
```

I'm not familiar with FakeFS so maybe there is a configuration option that I'm missing?

### CODE REVIEW

**Comment by etiennebarrie** [comment]
```ruby
require "erb_lint/rake/setup"
```

I find that API/entrypoint a bit weird. For comparison, here's RDoc:
```ruby
require "rdoc/task"
RDoc::Task.new(:rdoc)
```
Bundler gem tasks:
```ruby
require "bundler/gem_tasks"
```
Rubocop:
```ruby
require "rubocop/rake_task"
RuboCop::RakeTask.new
```

I think it would be useful to be able to configure the task, without having to pass arguments on the command-line. Ideally you want the task to be pre-configured so that you can add it as one of the `default` task dependencies and be done with it.

```ruby
task default: [:test, :rubocop, :erb_lint]
```

Also Rake's handling of arguments with the `[]` on the command-line is pretty peculiar, though it works, but from another task, it's pretty annoying, as they can't seem to be used as dependencies, you have to invoke them programmatically, something like:

```ruby
task default: [:test, :rubocop, :my_erb_lint]
task :my_erb_lint do
  Rake::Task[:erb_lint].invoke("--config", "config/erb-lint.yml", "--lint-all")
end
```

Instead we should have something more similar to Rubocop where you call new (and you can choose the name of the task), then you can pass configuration to the task (I think just passing an array that acts as ARGV is fine for that).

**Comment by bkuhlmann** [comment]
Thanks. I've incorporated your feedback by rebasing all changes (see the README for further details). I'll respond to your notes in the order defined:

- Yeah, I suppose `require "erb_lint/rake/setup"` is strange if you've never encountered it before. I first saw it with Bundler/Thor ~13 years back but I don't think they use this anymore (I might also be thinking of the wrong gems -- or they've refactored since -- so I couldn't point to them directly).
- Yes, I opted to follow the RDoc/Reek pattern with the new corrections.
- Both the original implementation and now the new implementation support the ability to define `erb_lint` as a default task.
- Yeah, I've always thought Rake's CLI was a bit fiddly and could be better (I wish that was the case).

File: README.md [hunk 1]
```diff
@@ -561,6 +561,25 @@ def autocorrect(_processed_source, offense)
 end
 ```
 
+## Rake
+
+You can add Rake support by adding the following to your `Rakefile`:
+
+    require "erb_lint/rake/task"
```
Comment (etiennebarrie) [inline README.md:568]: Can we change to `erb_lint/rake_task`?

File: lib/erb_lint/rake/task.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+require "rake"
+require "rake/tasklib"
+require "erb_lint/cli"
+
+module ERBLint
+  module Rake
+    # Defines Rake tasks for use.
+    class Task < ::Rake::TaskLib
+      LOGGER = Logger.new($stdout, formatter: ->(_severity, _at, _name, message) { "#{message}\n" })
```
Comment (etiennebarrie) [inline lib/erb_lint/rake/task.rb:11]: I don't think this is necessary. Let's just use `puts` instead, or maybe [`Rake.rake_output_message`](https://ruby.github.io/rake/Rake/FileUtilsExt.html#method-i-rake_output_message) ?

File: spec/lib/rake/task_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+require "erb_lint/rake/task"
```
Comment (etiennebarrie) [inline spec/lib/rake/task_spec.rb:4]: It seems like there's a conflict between `Rake` and `FakeFS`. Rake adds methods to `FileUtils` and then calls them (at least `commands`). It looks like you can work around that this way:
```suggestion
FakeFS.deactivate!
require "erb_lint/rake/task"
FakeFS.activate!
```

Comment (bkuhlmann) [reply spec/lib/rake/task_spec.rb]: Fixed. Thanks, the full test suite runs now.

File: README.md [hunk 4]
```diff
@@ -561,6 +561,25 @@ def autocorrect(_processed_source, offense)
...
+    require "erb_lint/rake_task"
+    ERBLint::RakeTask.new
+
+The Rake task can be customized as follows:
+
+    # Custom arguments.
+    ERBLint::RakeTask.new { %w[--autocorrect .] }
+
+    # Custom name and arguments.
+    ERBLint::RakeTask.new(:erb_auto_correct) { %w[--autocorrect .] }
```
Comment (etiennebarrie) [inline README.md:577]: I'm not super convinced about passing arguments as the value yielded by the block. What do you think about this?
```suggestion
    ERBLint::RakeTask.new(:erb_auto_correct, args: %w[--autocorrect .])
```

**Comment by bkuhlmann** [comment]
Hey Rafael, thanks. Make sense. I'll close this down in favor of inllining the task. :bow:

**Pull request closed without merging**